packages/core/src/routes/interaction/verifications/mfa-payload-verification.ts

```diff
@@ -128,6 +130,26 @@ const verifyBindBackupCode = async (
   return { type, codes };
 };
 
+const verifyBackupCode = async (
+  mfaVerifications: User['mfaVerifications'],
+  payload: BackupCodeVerificationPayload
+): Promise<VerifyMfaResult> => {
+  const backupCode = mfaVerifications.find(
+    (mfa): mfa is MfaVerificationBackupCode => mfa.type === MfaFactor.BackupCode
+  );
+
+  assertThat(backupCode, 'session.mfa.invalid_backup_code');
+
+  assertThat(
+    backupCode.codes.some((code) => code.code === payload.code && !code.usedAt),
+    'session.mfa.invalid_backup_code'
+  );
```

Comment:
```suggestion
  assertThat(
    backupCode.codes?.some((code) => code.code === payload.code && !code.usedAt),
    'session.mfa.invalid_backup_code'
  );
```

Reply:
Have to split into 2 assertions, otherwise `backupCode` can be undefined. Added a comment.